### PR TITLE
Coverage

### LINKED ISSUE #214: Increase test-coverage
We have decent test-coverage with recent code but there is some debt, mostly with early code (queries, property queries). Also, with null-safety enabled, there are now distinct code-branches in the generator for nullable and non-nullable fields so we should test both (current test entities are all nullable). We should:
* [x] introduce test-coverage report to CI - is that possible with dart tooling?
* [x] add nullable entities and cover them with tests
* [x] find blank spaces and cover them (as said above, I suspect that's mostly queries).

### PR DESCRIPTION
**Pull request by vaind**
closes #214

### CODE REVIEW

File: objectbox/test/boxnn_test.dart [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  setUp(() {
+    env = TestEnv('box-nn');
+    box = env.store.box();
+  });
+
+  tearDown(() => env.close());
+
+  test('CRUD', () {
+    final objects = [
+      TestEntityNN(),
```
Comment (greenrobot-team) [inline objectbox/test/boxnn_test.dart:22]: Maybe change the `NN` (and `nn`) to at least `NonNull` or similar? Was kind of confused by this as I thought of N as a number placeholder.

File: objectbox/test/boxnn_test.dart [hunk 2]
```diff
@@ -0,0 +1,88 @@
+import 'dart:io';
+import 'dart:typed_data';
+
+import 'package:test/test.dart';
+import 'package:objectbox/objectbox.dart';
+import 'entity.dart';
+import 'entity2.dart';
+import 'test_env.dart';
+
+// Tests non-nullable fields
```
Comment (greenrobot-team) [inline objectbox/test/boxnn_test.dart:8]: Maybe add a note that this is basically a copy of the regular box tests?

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -32,4 +32,35 @@ jobs:
...
+          sudo apt-get install lcov
+          make depend
+          make coverage
+      - uses: devmasx/coverage-check-action@v1.2.0
+        with:
+          type: lcov
+          result_path: objectbox/coverage/lcov.info
+          min_coverage: 90
+          token: ${{ github.token }}
+      - uses: actions/setup-dotnet@v1
```
Comment (greenrobot-team) [inline .github/workflows/lint.yml:54]: https://github.com/actions/virtual-environments/blob/ubuntu20/20210614.1/images/linux/Ubuntu2004-README.md#net-core-sdk says the latest .NET Core SDK is always installed. No? Edit: k, I guess just copy/pasted from ReportGenerator.

**Review by greenrobot-team — COMMENTED** [review]
Simple enough, LGTM.

File: objectbox/test/boxnn_test.dart (hunk 2 above)
Comment (vaind) [reply objectbox/test/boxnn_test.dart:8]: but it is not really, it's much simplified here. added the comment to datetime field test which is pretty much the same

File: objectbox/test/boxnn_test.dart (hunk 1 above)
Comment (vaind) [reply objectbox/test/boxnn_test.dart]: Ok

File: .github/workflows/lint.yml (hunk 3 above)
Comment (vaind) [reply .github/workflows/lint.yml]: seems to work after removing, thx for the suggestion

**Pull request merged**